chia/protocols/harvester_protocol.py

```diff
@@ -60,6 +79,7 @@ class RequestSignatures(Streamable):
     challenge_hash: bytes32
     sp_hash: bytes32
     messages: List[bytes32]
+    message_data: Optional[List[SignatureRequestSourceData]]
```

Comment:
would it not be simpler to make this field a `SignatureRequestSourceData` (instead of a list of them), and lift the restriction of just having one field set.

Don't we want to send all the source data anyway?

Reply:
excuse my ignorance, but couldn't this just be:
```
    foliage_block_data: Optional[FoliageBlockData]
    foliage_transaction_block_data: Optional[FoliageTransactionBlock]
```

as well? Or does the harvester sign more than those two blocks?

Reply:
In most cases the harvester does sign two messages at once.

Reply:
The Harvester is able to sign hashes belonging to serialized versions of all of the fields listed in `SignatureRequestSourceData`. I added all the known types it signs in that list. First signature requested is for the signage points (which can be VDFInfos or one of the 2 SubSlot ones), then, potentially, a Partial, finally FoliageBlockData, which could or could not include the transaction block.
They come in pairs usually, like Max said, except for the Partial which is sent by itself. It's in a List so that the data corresponding to the particular hash being signed can easily be obtained via the same index.

The fields are also separate even with the same data type (ie VDFInfo) so that the original purpose of data can be identified via its field location